packages/hash/api/src/db/postgres/util.ts

```diff
@@ -8,120 +7,3 @@ export const mapColumnNamesToSQL = (columnNames: string[], prefix?: string) =>
     ),
     sql`, `,
   );
-
-const isObject = (val: any) => typeof val === "object" && val !== null;
-
-const isObjectOrArray = (val: any) => Array.isArray(val) || isObject(val);
-
-// @todo: it would be preferrable to import the __linkedData type definition from
-// adapter.ts to avoid duplication. However, that type includes `entityTypeId` which
-// is not currently set by the links in "Block" entities. Return here when we refactor
-// the definition of Block entities to use the __linkedData concept.
-export type Link = {
-  entityId: string;
-  entityVersionId?: string;
-};
-
-/**
- * Validates a `__linkedData` field.
- * @returns a `Link` if it's valid
- * @throws if it's invalid
- */
-const getLink = (linkedData: any): Link | null => {
-  if (!isObject(linkedData)) {
-    throw new Error("must be an object");
-  }
-  if (linkedData.aggregate) {
-    // We can ignore linked data based on an aggregation here
-    return null;
-  }
-  if (!linkedData.entityId) {
-    throw new Error("field 'entityId' missing");
-  }
-  if (!(typeof linkedData.entityId === "string")) {
-    throw new Error("field 'entityId' must be a string");
-  }
-  if (
-    linkedData.entityVersionId &&
-    !(typeof linkedData.entityVersionId === "string")
-  ) {
-    throw new Error("field 'entityVersionId', if set, must be a string");
-  }
-  return {
-    entityId: linkedData.entityId,
-    entityVersionId: linkedData.entityVersionId,
-  };
-};
-
-/**
- * Recursively traverse the properties of an entity, gathering the IDs of all entities
- * it references through the `__linkedData` field.
- * */
-export const gatherLinks = (entity: DbEntity): Link[] => {
```

Comment:
I think this can now be removed as we are no longer storing links in the `properties` of an entity (either using either the `__linkedData` field for generic entities, the `contents` field for pages, or the `entity` field for blocks)

Reply:
So we still use legacy links when text entities need containing entities (such as in the Header block or the Callout block). Does this break that? Need to look into it

Reply:
Sorry @nathggns , my first comment assumed this would break something, but uses of the `gatherLinks` method was removed a while ago. So this should just be removing dead code.

Good to know about legacy links still being used in the text entities though. Their functionality shouldn't be impacted by this PR, so if they were resolving before they should still resolve now in the same way. They just won't be picked up on by things like search on the backend, but I think that's okay.

Either way it'd be good to transition away from these legacy links soon so we can remove the `properties` resolver, and not have ambiguity about which part of the codebase has access to which links. To my surprise I couldn't find a task for this, so have created one here: https://app.asana.com/0/1201095311341924/1201850801682936/f

Reply:
Ah, thanks for explaining!